src/utils.ts

```diff
@@ -1717,15 +1722,35 @@ export class newText extends newShape {
   get fullFont() { return newText.buildFont(this.style, this.fontsize, this.font) }
 
   private automaticFontSize(context: CanvasRenderingContext2D): number {
-    let tmp_context: CanvasRenderingContext2D = context
+    let tmp_context: CanvasRenderingContext2D = context;
     let pxMaxWidth: number = this.width * this.scale;
     tmp_context.font = '1px ' + this.font;
     return Math.min(pxMaxWidth / (tmp_context.measureText(this.text).width), DEFAULT_FONTSIZE)
   }
 
   public buildPath(): Path2D {
     const path = this.path;
-    path.rect(this.origin.x, this.origin.y, this.width, this.fontsize);
+    let origin = this.origin.copy();
+    let height = this.height;
+    switch (this.align) {
+      case "center": origin.x -= this.width / 2; break
```

Comment:
This could be a function using if and return statement

Reply:
Thanks and done !